src/runner/bootstrapper.js

```diff
@@ -21,48 +22,67 @@ export default class Bootstrapper {
         this.reporter = null;
     }
 
-    static async _convertAliasOrPathToBrowserInfo (browser, installations) {
-        if (typeof browser === 'string') {
-            if (browser in installations)
-                return installations[browser];
+    static async _convertPathAndCmdToBrowser (browserArg) {
+        var pathProvider = new BrowserProvider('path');
+        var options      = browserArg.cmd ? { cmd: browserArg.cmd } : null;
 
-            var pathDescriptorMatch = BROWSER_PATH_DESCRIPTOR_RE.exec(browser);
+        return new Browser(pathProvider, browserArg.path, options);
+    }
+
+    static async _convertAliasToBrowser (browserArg) {
+        var providerRegExpMatch = BROWSER_PROVIDER_RE.exec(browserArg);
+
+        if (!providerRegExpMatch)
+            throw new GeneralError(MESSAGE.cantFindBrowser, browserArg);
 
-            if (!pathDescriptorMatch)
-                throw new GeneralError(MESSAGE.cantFindBrowser, browser);
+        var providerName = providerRegExpMatch[1] || 'local';
+        var alias        = providerRegExpMatch[2];
 
-            var path        = pathDescriptorMatch[1];
-            var browserInfo = await getBrowserInfo(path);
+        var provider = new BrowserProvider(providerName);
 
-            if (!browserInfo)
-                throw new GeneralError(MESSAGE.cantFindBrowser, browser);
+        if (alias === 'all') {
+            var allAliases = await provider.listAvailableAliases();
 
-            return browserInfo;
+            return allAliases.map(al => new Browser(provider, al));
         }
 
-        return browser;
+        if (!await provider.isValidAlias(alias))
+            throw new GeneralError(MESSAGE.cantFindBrowser, browserArg);
+
+        return new Browser(provider, alias);
+    }
+
+    static async _getBrowser (browserArg) {
+        if (browserArg instanceof BrowserConnection)
+            return browserArg;
+
+        if (browserArg && browserArg.path)
+            return Bootstrapper._convertPathAndCmdToBrowser(browserArg);
+
+        if (typeof browserArg === 'string')
+            return Bootstrapper._convertAliasToBrowser(browserArg);
+
+        throw new GeneralError(MESSAGE.cantFindBrowser, browserArg);
     }
 
-    _createConnectionFromBrowserInfo (browserInfo) {
-        if (browserInfo instanceof BrowserConnection)
-            return browserInfo;
+    _createConnectionForBrowser (browser) {
+        if (browser instanceof BrowserConnection)
+            return browser;
 
-        return new LocalBrowserConnection(this.browserConnectionGateway, browserInfo);
+        return new BrowserConnection(this.browserConnectionGateway, browser);
     }
 
-    async _getBrowserInfo () {
+    async _getBrowsers () {
```

Comment:
Since we have `browser` entity now we should implictily call it `_getBrowserConnections`. Also in general we try to avoid names like `_getBrowser` and `_getBrowsers`, because it' may be confusing. Name it `_getBrowserConnectionList` instead